tests/func/test_repro.py

```diff
@@ -841,430 +818,6 @@ def test(self):
-        cache = (
-            self.scheme
-            + self.scheme_sep
-            + self.bucket
-            + self.sep
-            + str(uuid.uuid4())
-        )
-
-        ret = main(["config", "cache." + self.cache_scheme, "myrepo"])
-        self.assertEqual(ret, 0)
-        ret = main(["remote", "add", "myrepo", cache])
-        self.assertEqual(ret, 0)
-        if self.cache_type:
-            ret = main(["remote", "modify", "myrepo", "type", self.cache_type])
-            self.assertEqual(ret, 0)
-
-        remote_name = "myremote"
-        remote_key = str(uuid.uuid4())
-        remote = (
-            self.scheme + self.scheme_sep + self.bucket + self.sep + remote_key
-        )
-
-        ret = main(["remote", "add", remote_name, remote])
-        self.assertEqual(ret, 0)
-        if self.cache_type:
-            ret = main(
-                ["remote", "modify", remote_name, "type", self.cache_type]
-            )
-            self.assertEqual(ret, 0)
-
-        self.dvc = DvcRepo(".")
-
-        foo_key = remote_key + self.sep + self.FOO
-        bar_key = remote_key + self.sep + self.BAR
-
-        foo_path = (
-            self.scheme + self.scheme_sep + self.bucket + self.sep + foo_key
-        )
-        bar_path = (
-            self.scheme + self.scheme_sep + self.bucket + self.sep + bar_key
-        )
-
-        # Using both plain and remote notation
-        out_foo_path = "remote://" + remote_name + "/" + self.FOO
-        out_bar_path = bar_path
-
-        self.write(self.bucket, foo_key, self.FOO_CONTENTS)
-
-        import_stage = self.dvc.imp_url(out_foo_path, "import")
-
-        self.assertTrue(os.path.exists("import"))
-        self.assertTrue(filecmp.cmp("import", self.FOO, shallow=False))
-        self.assertEqual(self.dvc.status([import_stage.path]), {})
-        self.check_already_cached(import_stage)
-
-        import_remote_stage = self.dvc.imp_url(
-            out_foo_path, out_foo_path + "_imported"
-        )
-        self.assertEqual(self.dvc.status([import_remote_stage.path]), {})
-
-        cmd_stage = self._run(
-            outs=[out_bar_path],
-            deps=[out_foo_path],
-            cmd=self.cmd(foo_path, bar_path),
-            name="external-base",
-            external=True,
-        )
-
-        self.assertEqual(self.dvc.status([cmd_stage.addressing]), {})
-        self.assertEqual(self.dvc.status(), {})
-        self.check_already_cached(cmd_stage)
-
-        self.write(self.bucket, foo_key, self.BAR_CONTENTS)
-
-        self.assertNotEqual(self.dvc.status(), {})
-
-        self.dvc.update([import_stage.path])
-        self.assertTrue(os.path.exists("import"))
-        self.assertTrue(filecmp.cmp("import", self.BAR, shallow=False))
-        self.assertEqual(self.dvc.status([import_stage.path]), {})
-
-        self.dvc.update([import_remote_stage.path])
-        self.assertEqual(self.dvc.status([import_remote_stage.path]), {})
-
-        stages = self.dvc.reproduce(cmd_stage.addressing)
```

Comment:
Got rid of the repro test in favor of add/import/update/run(no-exec) tests because running commands is not trivial in a unified way, as there is no mechanism of translating our remote:// notation to a user-consumable one. So let's not have it for now (we are not loosing much) until https://github.com/iterative/dvc/issues/3920 is looked into, as this problem again illustrates the need for proper workspaces.